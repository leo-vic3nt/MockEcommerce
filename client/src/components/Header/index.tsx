import Brand from "@/components/Header/Brand/Brand.tsx";
import NavigationMenu from "@/components/Header/NavigationMenu/navigationMenu.tsx";
import FavoritesTracker from "@/components/Header/FavoritesTracker/FavoritesTracker.tsx";
import CartIcon from "@/components/Header/CartIcon/CartIcon.tsx";
import useMediaQuery from "@/hooks/useMediaQuery.tsx";
import { isDesktopMediaQueryString } from "@/lib/utils.ts";

function Header() {
	const isDesktop = useMediaQuery(isDesktopMediaQueryString);

	return (
		<div className="mx-auto flex items-center justify-between">
			<Brand withLogoImg={true} />
			{isDesktop && <NavigationMenu type="desktop" />}
			<div className="flex items-center gap-4 pr-5 lg:pt-3">
				{!isDesktop && <NavigationMenu type="mobile" />}
				<FavoritesTracker />
				<CartIcon />
			</div>
		</div>
	);
}

export default Header;
