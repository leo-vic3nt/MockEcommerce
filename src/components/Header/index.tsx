// @todo Move the brand component here, create the navbar using shadcn menubar and finally the cart, and search

import Brand from "@/components/Header/Brand/Brand.tsx";
import NavigationMenu from "@/components/Header/NavigationMenu/NavigationMenu.tsx";

function Header() {
	return (
		<div className="flex items-center gap-96">
			<Brand withLogoImg={true} />
			<NavigationMenu />
		</div>
	);
}

export default Header;
