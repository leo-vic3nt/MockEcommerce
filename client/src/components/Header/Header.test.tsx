import { describe, expect, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "@/components/Header/index.tsx";

vi.mock("@/hooks/useMediaQuery", () => ({
	default: vi.fn(() => true),
}));

describe("Header", () => {
	it("matches snapshot", () => {
		const { container } = render(
			<MemoryRouter>
				<Header />
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});
});
