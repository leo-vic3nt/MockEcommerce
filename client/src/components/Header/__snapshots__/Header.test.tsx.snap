// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
<div>
  <header
    class="mx-auto flex items-center justify-between"
  >
    <a
      data-discover="true"
      href="/"
    >
      <div
        class="flex flex-nowrap items-center justify-center"
      >
        <img
          alt="logo image"
          class="-mr-1.5 size-10 lg:size-14 pt-1 xl:size-16"
          src="/src/assets/icons/logo.svg"
        />
        <h1
          class="whitespace-nowrap font-cooperBlack text-2xl lowercase lg:text-3xl xl:text-5xl"
        >
          nice orange
          <span
            class="text-orange-contrast"
          >
            .
          </span>
        </h1>
      </div>
    </a>
    <nav
      class="text-md absolute left-1/2 flex -translate-x-1/2 gap-10 text-zinc-800 lg:pt-3 xl:text-xl"
    >
      <a
        aria-current="page"
        class="font-bold after:w-full after:bg-orange-500 relative leading-tight after:absolute after:bottom-0 
				after:left-0 after:h-0.5 after:w-0 after:transition-[width] after:duration-300 hover:after:w-full"
        data-discover="true"
        href="/"
      >
        Home
      </a>
      <a
        class="false after:bg-orange-500 relative leading-tight after:absolute after:bottom-0 
				after:left-0 after:h-0.5 after:w-0 after:transition-[width] after:duration-300 hover:after:w-full"
        data-discover="true"
        href="/shop"
      >
        Shop
      </a>
      <a
        class="false after:bg-orange-500 relative leading-tight after:absolute after:bottom-0 
				after:left-0 after:h-0.5 after:w-0 after:transition-[width] after:duration-300 hover:after:w-full"
        data-discover="true"
        href="/about"
      >
        About
      </a>
    </nav>
    <div
      class="flex items-center gap-4 pr-5 lg:pt-3"
    >
      <svg
        class="lucide lucide-heart fill-orange-contrast stroke-orange-contrast size-6 lg:size-8"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
        />
      </svg>
      <svg
        class="lucide lucide-shopping-bag stroke-orange-contrast size-6 lg:size-8"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
        />
        <path
          d="M3 6h18"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
        />
      </svg>
    </div>
  </header>
</div>
`;
